### ISSUE TITLE
GM_deleteValue followed by GM_getValue returns the value as it was on page load.

### ISSUE THREAD

**Issue by RobertZenz** [erosman/support#485]
Deleting a value with `GM_deleteValue` reverts it to the value as it was on page load.

Assume the following UserScript:

```javascript
// ==UserScript==
// @name TEST
// @grant GM_deleteValue
// @grant GM_getValue
// @grant GM_setValue
// ==/UserScript==

let setButton = document.createElement("button");
setButton.innerText = "Set";
setButton.addEventListener("click", function(mouseEvent) {
	GM_setValue("TEST", { test: "SomeValue" });
});

let deleteButton = document.createElement("button");
deleteButton.innerText = "Delete";
deleteButton.addEventListener("click", function(mouseEvent) {
	GM_deleteValue("TEST");
});

let showButton = document.createElement("button");
showButton.innerText = "Show";
showButton.addEventListener("click", function(mouseEvent) {
	alert(JSON.stringify(GM_getValue("TEST")));
});

document.body.appendChild(setButton);
document.body.appendChild(deleteButton);
document.body.appendChild(showButton);
```

This adds three buttons to a page, to reproduce the problem:

 1. Hit "Set"
 2. Hit "Show" to verify that the value was set
 3. Reload the page
 4. Hit "Show" to verify that the value is still set
 5. Hit "Delete"
 6. Hit "Show" and see that the value is still set (actually, it reverted to the value as it was on page load)
 7. Reload the page
 8. Hit "Show" and see that the value is now gone as it should be

**Comment by RobertZenz** [comment]
It's not limited to arrays/objects, as I just realized.

**Comment by erosman** [comment]
I am traveling but I will check it as soon as I am able to.

**Comment by erosman** [comment]
Please try this and let me know:

```js
// ==UserScript==
// @name  485 TEST
// @match https://example.com/*
// @grant GM_deleteValue
// @grant GM_getValue
// @grant GM_setValue
// ==/UserScript==

const pre = document.createElement('pre');

const setButton = document.createElement('button');
setButton.textContent = 'Set';
setButton.addEventListener('click', () => GM_setValue('TEST', {test: Date.now()}));

const deleteButton = document.createElement('button');
deleteButton.textContent = 'Delete';
deleteButton.addEventListener('click', () => GM_deleteValue('TEST'));

const showButton = document.createElement('button');
showButton.textContent = 'Show';
showButton.addEventListener('click', () => pre.textContent += JSON.stringify(GM_getValue('TEST')) + '\n');

document.body.appendChild(setButton);
document.body.appendChild(deleteButton);
document.body.appendChild(showButton);
document.body.appendChild(pre);
```

**Comment by RobertZenz** [comment]
Same behavior:

```
    Page load
{"test":1654162393109}
    Set pressed
{"test":1654162399726}
    Delete pressed
{"test":1654162393109}
```

At this point I should maybe add that I'm using Waterfox G4.1.2.1 (based on Firefox 91 I believe, which is ESR), but I don't think that that should make a difference.

**Comment by erosman** [comment]
It is not like that for me.


new page
show
set + show
delete + show
```
undefined
{"test":1654166618654}
undefined
```
Reload tab + show
```
undefined
```

**Comment by RobertZenz** [comment]
You need to reload before deletion but after setting, as it does revert to the state as on page load.

 * New page
 * Set
 * Show
 * **Reload**
 * Show
 * Set
 * Show
 * Delete
 * Show

**Comment by erosman** [comment]
I see... let me check what is going on.

**Comment by erosman** [comment]
Found it... fixed for v2.57

**Comment by erosman** [comment]
v2.57 uploaded
